rpcclient/infrastructure.go

```diff
@@ -892,7 +929,7 @@ out:
 		// Send any messages ready for send until the shutdown channel
 		// is closed.
 		select {
-		case jReq := <-c.highPriorityPostQueue:
+		case jReq := <-c.nextPostRequest():
```

Comment:
This now no longer considers the `c.shutdown` channel. Because we'll block on the method call `c.nextPostRequest()` and then only actually `select` once that returns. So I think we need to consider `c.shutdown` within `nextPostRequest()` and then we can simply return the next request (instead of a channel) or a boolean that indicates we're shutting down.

Reply:
good catch, fixed!